stdlib/test/builtin/test_math.mojo

```diff
@@ -63,12 +63,12 @@ def test_max():
 
 
 def test_round():
-    assert_equal(0, round(0.0))
-    assert_equal(1, round(1.0))
-    assert_equal(1, round(1.1))
-    assert_equal(2, round(1.5))
-    assert_equal(2, round(1.9))
-    assert_equal(2, round(2.0))
+    assert_equal(0.0, round(0.0))
+    assert_equal(1.0, round(1.0))
+    assert_equal(1.0, round(1.1))
+    assert_equal(2.0, round(1.5))
+    assert_equal(2.0, round(1.9))
+    assert_equal(2.0, round(2.0))
 
```

Comment:
There's a couple of tests that had to be modified. They were finding the path of least resistance through the `String` overload of `assert_equal()`. If `String` was `EqualityComparable`, those overloads could be removed and this could be avoided.

Reply:
(when you add the `.0`, it uses the correct overload)

Reply:
Is this still the case on latest nightly?

Reply:
Yes, it's because there is an overload of assert equal that takes strings.

Reply:
Removing them right now causes problems because some tests use `assert_equal(StringLiteral, StringSlice[lifetime])` for example.

Reply:
Maybe those could call `str()` explicity.

Reply:
Also, there's lots of strange symbols in the `test_string_iter()` which are really confusing the language server.

Reply:
I kind of like what it does to the text colors :]
but maybe there's a better way.